internal/controller/awsaccount_controller.go

```diff
@@ -29,6 +31,37 @@ import (
 	kuadrav1 "github.com/Kuadrant/kuadra/api/v1"
 )
 
+func contains[T comparable](slice []T, val T) bool {
+	for _, element := range slice {
+		if element == val {
+			return true
+		}
+	}
+	return false
+}
+
+func indexOf[T comparable](slice []T, element T) int {
+	for i, val := range slice {
+		if val == element {
+			return i
+		}
+	}
+	return -1
+}
+
+func remove[T any](slice []T, i int) []T {
+	return append(slice[:i], slice[i+1:]...)
+}
+
+func getLeftDifference[T comparable](left []T, right []T) (leftDifference []T) {
+	for _, val := range left {
+		if !contains[T](right, val) {
+			leftDifference = append(leftDifference, val)
+		}
+	}
+	return leftDifference
+}
+
```

Comment:
Nice use of generics! I'd probably suggest to move these functions to a utility package as they'll probably be useful to be called from other controllers